test/integration/auth/auth.prose.test.ts

```diff
@@ -239,26 +254,31 @@ describe('Authentication Spec Prose Tests', function () {
             authSource: 'admin'
           };
 
-          const noUserClient = this.configuration.newClient({}, noUsernameOptions);
+          try {
+            this.configuration.newClient({}, noUsernameOptions);
+            expect.fail(
+              'Creating a new client with a password and no username must fail validation.'
+            );
+          } catch (noUserError) {
+            // NOTE: This prose test fails Node's validation of the credentials object - a username
+            // MUST always be provided but it satisfies the test requirement of not getting a
+            // network or command error on the handshake.
+            expect(noUserError).to.match(/username/);
+          }
           const badPasswordClient = this.configuration.newClient({}, badPasswordOptions);
-          const noUserError = await noUserClient
-            .db('test')
-            .stats()
-            .catch(e => e);
           const badPasswordError = await badPasswordClient
             .db('test')
             .stats()
             .catch(e => e);
-          expect(noUserError).to.match(/Authentication failed/);
           expect(badPasswordError).to.match(/Authentication failed/);
-          await noUserClient.close();
           await badPasswordClient.close();
         }
-      });
+      );
 
-      it('should send speculativeAuthenticate on initial handshake on MongoDB 4.4+', {
-        metadata: { requires: { mongodb: '>=4.4', topology: ['single'] } },
-        test: async function () {
+      it(
+        'sends speculativeAuthenticate on initial handshake on MongoDB 4.4+',
+        metadata,
+        async function () {
```

Comment:
I had to read the spec here to clarify:

> Older servers will ignore the speculativeAuthenticate argument. New servers will participate in the standard authentication conversation if this argument is missing.

I don't think we have any branching logic determining when we use speculative authentication.  Maybe we should remove the server version from the test?

Reply:
I've removed the 4.4+ version from the description but left the metadata there as all tests still need 3.7.3+.